server/methods/catalog.js

```diff
@@ -829,12 +906,22 @@ Meteor.methods({
       return ids;
     });
 
-    Products.remove({
-      _id: {
-        $in: ids
-      }
+    const options = { userId: this.userId };
+    let shouldRemoveProduct;
+    ids.forEach((_id) => {
```

Comment:
Should use `id.every()` since its a loop and the last thing to return true will make line 915 execute.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every